src/arch/src/x86_64/msr.rs

```diff
@@ -282,20 +282,44 @@ pub fn create_boot_msr_entries() -> Vec<kvm_msr_entry> {
     ]
 }
 
+/// Error type for [`set_msrs`].
+#[derive(Debug, thiserror::Error)]
+pub enum SetMSRsError {
+    /// Failed to create [`vmm_sys_util::fam::FamStructWrapper`] for MSRs.
+    #[error("Could not create `vmm_sys_util::fam::FamStructWrapper` for MSRs")]
```

Comment:
The issue with giving further context here is that since `fam::Error` doesn't implement `std::error::Error` we can't display it nicely (I recently got a PR merged fixing this https://github.com/rust-vmm/vmm-sys-util/pull/168 but we would need to update the version for this).

In this context I'd argue that we leave it in the most descriptive state, considering this will form an error message like:
> Failed to boot microVM: Failed to initialize vCPU: Failed to set MSRs: Could not create `vmm_sys_util::fam::FamStructWrapper` for MSRs

When we update our version we can update here to add further context.

It would be useful to remind someone in the future to add this context. Do you know if there is there any recommended approach here?

Reply:
> In this context I'd argue that we leave it in the most descriptive state

My concern is that for a Firecracker user, this error isn't descriptive. Parameterising a more specific message from where the error occurs so that the user can troubleshoot would be a better way to go here.

Reply:
I think this is as specific as we can go, all we know from `Msrs::from_entries(&msr_entries)` is that an error occurred. `fam::Error` contains this description, but for now, we cannot display it (we need there to be a new version of the crate containing `fam::Error` so as to expose the `std::error::Error` implementation, so we can then display it).

While it would be ideal that every error would be descriptive enough to highlight solutions I think for the moment here it cannot be well done and this is still an improvement over the existing even less specific error.

Reply:
I think the context of the error here might clarify the usefulness of it. A less specific but general error like "Internal error" for example would at least indicate a system error versus a user error. The current description is for a developer's use, not for the user of Firecracker.

Reply:
> A less specific but general error like "Internal error" for example would at least indicate a system error versus a user error. The current description is for a developer's use, not for the user of Firecracker.

I think this distinction is difficult to draw and somewhat unnecessary. If we have a descriptive error we can then on the front-end present the user with a simple `"Internal error"`. I do not see any argument for having a less descriptive error (in the same sense I would not see an argument for having less descriptive documentation).

Apologies if I am misunderstanding.